test/module/SmartContractOwnershipRegistry.Module.specs.ts

```diff
@@ -0,0 +1,491 @@
+            await saFactory.deployCounterFactualAccount(smartContractOwnershipRegistryModule.address,smartContractOwnershipSetupData,smartAccountDeploymentIndex+3);
+            const unregisteredSA = await hre.ethers.getContractAt("SmartAccount", unregisteredSmartAccountAddress);
+
+            //fund the smart account
+            await mockToken.mint(unregisteredSmartAccountAddress, ethers.utils.parseEther("1000").toString());
+            await deployer.sendTransaction({
+                to: unregisteredSmartAccountAddress,
+                value: ethers.utils.parseEther("60"),
+            });
+            //renounce ownership
+            const renounceOwnershipUserOp = await makeSARegistryModuleUserOp(
+                "executeCall",
+                [smartContractOwnershipRegistryModule.address,0,smartContractOwnershipRegistryModule.interface.encodeFunctionData("renounceOwnership",[])],
+                unregisteredSmartAccountAddress,
+                baseSmartAccountOwner1,
+                entryPoint,
+                smartContractOwnershipRegistryModule.address,
+                ecdsaRegistryModule.address
+            );
+            const handleOpsTx = await entryPoint.handleOps([renounceOwnershipUserOp],alice.address);
+            await expect(handleOpsTx).to.not.emit(entryPoint, "UserOperationRevertReason");
+            await expect(smartContractOwnershipRegistryModule.getOwner(unregisteredSA.address)).to.be.revertedWith("NoOwnerRegisteredForSmartAccount");
+
+            const sendTokenUserOp = await makeSARegistryModuleUserOp(
+                "executeCall",
+                [mockToken.address, 0, encodeTransfer(bob.address,tokenAmountToTransfer.toString())],
+                unregisteredSmartAccountAddress,
+                baseSmartAccountOwner1,
+                entryPoint,
+                smartContractOwnershipRegistryModule.address,
+                ecdsaRegistryModule.address
+            );
+
+            const provider = entryPoint?.provider;
+            const chainId = await provider!.getNetwork().then((net) => net.chainId);
+            const userOpHash = getUserOpHash(sendTokenUserOp,entryPoint.address,chainId);
+            await expect(smartContractOwnershipRegistryModule.validateUserOp(sendTokenUserOp,userOpHash)).to.be.revertedWith("NoOwnerRegisteredForSmartAccount");
+
+            const unregisteredSABalanceBefore = await mockToken.balanceOf(unregisteredSA.address);
+            await expect(entryPoint.handleOps([sendTokenUserOp],alice.address)).to.be.revertedWith("FailedOp");
+            expect(await mockToken.balanceOf(bob.address)).to.equal(bobBalanceBefore);
+            expect(await mockToken.balanceOf(unregisteredSA.address)).to.equal(unregisteredSABalanceBefore);
+        });
+
+        it("Reverts when length of user.signature is less than 65 ", async()=>{
```

Comment:
this case is not needed. there's no functionality that checks signature length in SCOwnershipRegistry, so nothing to test here

if in the Smart Account that is owner of userSA there will be not an ECDSA module as validator, this test won't pass.

Reply:
Understood